lib/http-proxy/common.js

```diff
@@ -198,6 +198,41 @@ common.urlJoin = function() {
 };
 
 /**
+ * Rewrites or removes the domain of a cookie header
+ *
+ * @param {String|Array} Header
+ * @param {Object} Config, mapping of domain to rewritten domain.
+ *                 '*' key to match any domain, null value to remove the domain.
+ *
+ * @api private
+ */
+common.rewriteCookieDomain = function rewriteCookieDomain(header, config) {
+  if (Array.isArray(header)) {
+    return header.map(function (headerElement) {
+      return rewriteCookieDomain(headerElement, config);
+    });
+  }
+  return header.replace(/(;\s*domain=)([^;]+)/, function(match, prefix, previousDomain) {
```

Comment:
Lets cache this regex so it isnt created every time the function is called. Put `var domainRegex = ...` up top here and use it in this function.